TerminalLogger guidance has incorrect syntax

### Type of issue

Typo

### Description

The usage for `dotnet build` shows the syntax for Terminal Logger as ` [--tl [auto|on|off]]`, but it currently needs to be ` [--tl=[auto|on|off]]` - note the replacement of space with equals. 

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-build.md

### Document Version Independent Id

5e495ddd-140e-de5c-a793-d4a9f88be51e

### Article author

tdykstra


---
[Associated WorkItem - 183673](https://dev.azure.com/msft-skilling/Content/_workitems/edit/183673)

> ### Description
> The usage for `dotnet build` shows the syntax for Terminal Logger as ` [--tl [auto|on|off]]`, but it currently needs to be ` [--tl=[auto|on|off]]` - note the replacement of space with equals.

From my research in https://github.com/dotnet/sdk/issues/36921#issuecomment-1814170480, it has to be a `:`, not a `=`:
```
C:\Users\Jeroen\Downloads\Test>dotnet publish --tl:on
MSBuild version 17.8.3+195e7f5a3 for .NET
Restore complete (0,4s)
  Test succeeded (0,2s) → bin\Release\net8.0\publish\

Build succeeded in 0,8s

C:\Users\Jeroen\Downloads\Test>dotnet publish --tl=on
MSBuild version 17.8.3+195e7f5a3 for .NET
MSBUILD : error MSB1001: Unknown switch.
    Full command line: 'C:\Program Files\dotnet\sdk\8.0.100\MSBuild.dll -maxcpucount -verbosity:m -restore -target:Publish --property:_IsPublishing=true -property:Configuration=Release --tl=on -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\8.0.100\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\8.0.100\dotnet.dll'
  Switches appended by response files:
Switch: --tl=on

For switch syntax, type "MSBuild -help"
```

--- a/docs/core/tools/dotnet-build.md
+++ b/docs/core/tools/dotnet-build.md
@@ -1,7 +1,7 @@
 ---
 title: dotnet build command
 description: The dotnet build command builds a project and all of its dependencies.
-ms.date: 07/19/2023
+ms.date: 11/27/2023
 ---
 # dotnet build
 
@@ -23,7 +23,7 @@ dotnet build [<PROJECT>|<SOLUTION>] [-a|--arch <ARCHITECTURE>]
     [-p|--property:<PROPERTYNAME>=<VALUE>]
     [-r|--runtime <RUNTIME_IDENTIFIER>]
     [--self-contained [true|false]] [--source <SOURCE>]
-    [--tl [auto|on|off]] [--use-current-runtime, --ucr [true|false]]
+    [--tl:[auto|on|off]] [--use-current-runtime, --ucr [true|false]]
     [-v|--verbosity <LEVEL>] [--version-suffix <VERSION_SUFFIX>]
 
 dotnet build -h|--help
@@ -151,19 +151,7 @@ The project or solution file to build. If a project or solution file isn't speci
 
   The URI of the NuGet package source to use during the restore operation.
 
-- **`--tl [auto|on|off]`**
-
-  Specifies whether the *terminal logger* should be used for the build output. The default is `auto`, which first verifies the environment before enabling terminal logging. The environment check verifies that the terminal is capable of using modern output features and isn't using a redirected standard output before enabling the new logger. `on` skips the environment check and enables terminal logging. `off` skips the environment check and uses the default console logger.
-
-  The terminal logger shows you the restore phase followed by the build phase. During each phase, the currently building projects appear at the bottom of the terminal. Each project that's building outputs both the MSBuild target currently being built and the amount of time spent on that target. You can search this information to learn more about the build. When a project is finished building, a single "build completed" section is written for that captures:
-
-  - The name of the built project
-  - The target framework (if multi-targeted)
-  - The status of that build
-  - The primary output of that build (which is hyperlinked)
-  - Any diagnostics generated for that project
-
-  This option is available starting in .NET 8.
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 - **`-v|--verbosity <LEVEL>`**
 

--- a/docs/core/tools/dotnet-clean.md
+++ b/docs/core/tools/dotnet-clean.md
@@ -17,7 +17,8 @@ ms.date: 02/14/2020
 dotnet clean [<PROJECT>|<SOLUTION>] [-c|--configuration <CONFIGURATION>]
     [-f|--framework <FRAMEWORK>] [--interactive]
     [--nologo] [-o|--output <OUTPUT_DIRECTORY>]
-    [-r|--runtime <RUNTIME_IDENTIFIER>] [-v|--verbosity <LEVEL>]
+    [-r|--runtime <RUNTIME_IDENTIFIER>] [--tl:[auto|on|off]]
+    [-v|--verbosity <LEVEL>]
 
 dotnet clean -h|--help
 ```
@@ -59,6 +60,8 @@ The MSBuild project or solution to clean. If a project or solution file is not s
 * **`-r|--runtime <RUNTIME_IDENTIFIER>`**
 
   Cleans the output folder of the specified runtime. This is used when a [self-contained deployment](../deploying/index.md#publish-self-contained) was created.
+
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 [!INCLUDE [verbosity](../../../includes/cli-verbosity-normal.md)]
 

--- a/docs/core/tools/dotnet-pack.md
+++ b/docs/core/tools/dotnet-pack.md
@@ -1,7 +1,7 @@
 ---
 title: dotnet pack command
 description: The dotnet pack command creates NuGet packages for your .NET project.
-ms.date: 08/23/2021
+ms.date: 11/27/2023
 ---
 # dotnet pack
 
@@ -18,7 +18,7 @@ dotnet pack [<PROJECT>|<SOLUTION>] [-c|--configuration <CONFIGURATION>]
     [--force] [--include-source] [--include-symbols] [--interactive]
     [--no-build] [--no-dependencies] [--no-restore] [--nologo]
     [-o|--output <OUTPUT_DIRECTORY>] [--runtime <RUNTIME_IDENTIFIER>]
-    [-s|--serviceable] [-v|--verbosity <LEVEL>]
+    [-s|--serviceable] [--tl:[auto|on|off]] [-v|--verbosity <LEVEL>]
     [--version-suffix <VERSION_SUFFIX>]
 
 dotnet pack -h|--help
@@ -108,6 +108,8 @@ You can provide MSBuild properties to the `dotnet pack` command for the packing 
 - **`-s|--serviceable`**
 
   Sets the serviceable flag in the package. For more information, see [.NET Blog: .NET Framework 4.5.1 Supports Microsoft Security Updates for .NET NuGet Libraries](https://aka.ms/nupkgservicing).
+
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 [!INCLUDE [verbosity](../../../includes/cli-verbosity.md)]
 

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -21,7 +21,8 @@ dotnet publish [<PROJECT>|<SOLUTION>] [-a|--arch <ARCHITECTURE>]
     [--no-restore] [--nologo] [-o|--output <OUTPUT_DIRECTORY>]
     [--os <OS>] [-r|--runtime <RUNTIME_IDENTIFIER>]
     [--sc|--self-contained [true|false]] [--no-self-contained]
-    [-s|--source <SOURCE>] [--use-current-runtime, --ucr [true|false]]
+    [-s|--source <SOURCE>] [--tl:[auto|on|off]]
+    [--use-current-runtime, --ucr [true|false]]
     [-v|--verbosity <LEVEL>] [--version-suffix <VERSION_SUFFIX>]
 
 dotnet publish -h|--help
@@ -207,6 +208,8 @@ For more information, see the following resources:
 - **`-r|--runtime <RUNTIME_IDENTIFIER>`**
 
   Publishes the application for a given runtime. For a list of Runtime Identifiers (RIDs), see the [RID catalog](../rid-catalog.md). For more information, see [.NET application publishing](../deploying/index.md) and [Publish .NET apps with the .NET CLI](../deploying/deploy-with-cli.md). If you use this option, use `--self-contained` or `--no-self-contained` also.
+
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 [!INCLUDE [verbosity](../../../includes/cli-verbosity-minimal.md)]
 

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -211,11 +211,11 @@ For more information, see the following resources:
 
 [!INCLUDE [tl](../../../includes/cli-tl.md)]
 
-[!INCLUDE [verbosity](../../../includes/cli-verbosity-minimal.md)]
-
 - **`--use-current-runtime, --ucr [true|false]`**
 
   Sets the `RuntimeIdentifier` to a platform portable `RuntimeIdentifier` based on the one of your machine. This happens implicitly with properties that require a `RuntimeIdentifier`, such as `SelfContained`, `PublishAot`, `PublishSelfContained`, `PublishSingleFile`, and `PublishReadyToRun`. If the property is set to false, that implicit resolution will no longer occur.
+
+[!INCLUDE [verbosity](../../../includes/cli-verbosity-minimal.md)]
 
 - **`--version-suffix <VERSION_SUFFIX>`**
 

--- a/docs/core/tools/dotnet-restore.md
+++ b/docs/core/tools/dotnet-restore.md
@@ -20,8 +20,8 @@ dotnet restore [<ROOT>] [--configfile <FILE>] [--disable-build-servers]
     [--interactive] [--lock-file-path <LOCK_FILE_PATH>] [--locked-mode]
     [--no-cache] [--no-dependencies] [--packages <PACKAGES_DIRECTORY>]
     [-r|--runtime <RUNTIME_IDENTIFIER>] [-s|--source <SOURCE>]
-    [--use-current-runtime, --ucr [true|false]] [--use-lock-file]
-    [-v|--verbosity <LEVEL>]
+    [--tl:[auto|on|off]] [--use-current-runtime, --ucr [true|false]]
+    [--use-lock-file] [-v|--verbosity <LEVEL>]
 
 dotnet restore -h|--help
 ```
@@ -143,6 +143,8 @@ There are three specific settings that `dotnet restore` ignores:
 - **`-s|--source <SOURCE>`**
 
   Specifies the URI of the NuGet package source to use during the restore operation. This setting overrides all of the sources specified in the *nuget.config* files. Multiple sources can be provided by specifying this option multiple times.
+
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 - **`--use-current-runtime, --ucr [true|false]`**
 

--- a/docs/core/tools/dotnet-run.md
+++ b/docs/core/tools/dotnet-run.md
@@ -1,7 +1,7 @@
 ---
 title: dotnet run command
 description: The dotnet run command provides a convenient option to run your application from the source code.
-ms.date: 08/18/2021
+ms.date: 11/27/2023
 ---
 # dotnet run
 
@@ -19,7 +19,8 @@ dotnet run [-a|--arch <ARCHITECTURE>] [-c|--configuration <CONFIGURATION>]
     [--launch-profile <NAME>] [--no-build]
     [--no-dependencies] [--no-launch-profile] [--no-restore]
     [--os <OS>] [--project <PATH>] [-r|--runtime <RUNTIME_IDENTIFIER>]
-    [-v|--verbosity <LEVEL>] [[--] [application arguments]]
+    [--tl:[auto|on|off]] [-v|--verbosity <LEVEL>]
+    [[--] [application arguments]]
 
 dotnet run -h|--help
 ```
@@ -121,6 +122,8 @@ To run the application, the `dotnet run` command resolves the dependencies of th
 - **`-r|--runtime <RUNTIME_IDENTIFIER>`**
 
   Specifies the target runtime to restore packages for. For a list of Runtime Identifiers (RIDs), see the [RID catalog](../rid-catalog.md).
+
+[!INCLUDE [tl](../../../includes/cli-tl.md)]
 
 [!INCLUDE [verbosity](../../../includes/cli-verbosity-minimal.md)]
 

--- a/includes/cli-tl.md
+++ b/includes/cli-tl.md
@@ -1,7 +1,7 @@
 ---
 author: tdykstra
 ms.author: tdykstra
-ms.date: 08/12/2021
+ms.date: 11/27/2023
 ms.topic: include
 ---
 - **`--tl:[auto|on|off]`**

--- a/includes/cli-tl.md
+++ b/includes/cli-tl.md
@@ -1,0 +1,19 @@
+---
+author: tdykstra
+ms.author: tdykstra
+ms.date: 08/12/2021
+ms.topic: include
+---
+- **`--tl:[auto|on|off]`**
+
+  Specifies whether the *terminal logger* should be used for the build output. The default is `auto`, which first verifies the environment before enabling terminal logging. The environment check verifies that the terminal is capable of using modern output features and isn't using a redirected standard output before enabling the new logger. `on` skips the environment check and enables terminal logging. `off` skips the environment check and uses the default console logger.
+
+  The terminal logger shows you the restore phase followed by the build phase. During each phase, the currently building projects appear at the bottom of the terminal. Each project that's building outputs both the MSBuild target currently being built and the amount of time spent on that target. You can search this information to learn more about the build. When a project is finished building, a single "build completed" section is written that captures:
+
+  - The name of the built project.
+  - The target framework (if multi-targeted).
+  - The status of that build.
+  - The primary output of that build (which is hyperlinked).
+  - Any diagnostics generated for that project.
+
+  This option is available starting in .NET 8.

